forward_script/src/PageInformationCollector.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Wikimedia\ForwardScript;
+
+use Mediawiki\Api\MediawikiApi;
+use Mediawiki\Api\SimpleRequest;
+use Symfony\Component\Process\Process;
+use Symfony\Component\Process\Exception\RuntimeException;
+use Wikimedia\ForwardScript\ApplicationException;
+
+/**
+ * Get page content and categories from Wikipedia, return information about the page.
+ *
+ * @package Wikimedia\ForwardScript
+ */
+class PageInformationCollector {
+
+	/**
+	 * @var MediawikiApi
+	 */
+	protected $api;
+
+	/**
+	 * @var Process
+	 */
+	protected $processCommand;
+
+	/**
+	 * @var array
+	 */
+	protected $defaultCategories;
+
+	public function __construct( MediawikiApi $api, Process $processCommand, $defaultCategories=[] ) {
+		$this->api = $api;
+		$this->processCommand = $processCommand;
+		$this->defaultCategories = $defaultCategories;
+	}
+
+	public function getInformation( $pageTitle, $monumentId ) {
```

Comment:
This method does quite a lot of things. It could be broken down several methods for validation, running the command and finally returning the decoded output.